Switch Symbol Database to open beta (#23635)
DataDog/documentation: content/en/dynamic_instrumentation/symdb/dotnet.md

## Requirements

- [Dynamic Instrumentation][1] is enabled for your service.
- Tracing library [`dd-trace-dotnet`][1] 2.54.0 or higher is installed.

## Installation

Run your service with Dynamic Instrumentation enabled, and additionally enable autocomplete and search:

1. Set the `DD_SYMBOL_DATABASE_UPLOAD_ENABLED` environment variable to `true`.
1. Specify the `DD_SERVICE` and `DD_VERSION` [Unified Service Tags][2] as environment variables.
1. After starting your service with Dynamic Instrumentation and autocomplete and search enabled, you can use Dynamic Instrumentation's IDE-like features on the [**APM** > **Dynamic Instrumentation**][3] page.

## Additional configuration

### Third-party detection

If autocomplete suggestions do not appear for your package or module, it may be incorrectly recognized as third-party code. The autocomplete and search features use a heuristic to filter out third-party code, which can sometimes lead to accidental misclassification.

If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with the following settings:

```

> export DD_THIRD_PARTY_DETECTION_EXCLUDES=<LIST_OF_USER_CODE_PACKAGE_PREFIXES>

Reviewer: For .NET we use `DD_SYMBOL_DATABASE_THIRD_PARTY_DETECTION_EXCLUDES` and `DD_SYMBOL_DATABASE_THIRD_PARTY_DETECTION_INCLUDES`

Author: So, does .NET not support `DD_THIRD_PARTY_DETECTION_EXCLUDES` and `_INCLUDES`? 🤔 It should support them as default and only use `DD_SYMBOL_DATABASE_THIRD_PARTY_DETECTION_EXCLUDES` to augment the list or override.

---
title: Enable autocomplete and search for .NET
kind: Documentation
aliases:
    - /tracing/dynamic_instrumentation/enabling/dotnet/
is_beta: false
private: false
code_lang: dotnet
type: multi-code-lang
code_lang_weight: 30
further_reading:
    - link: 'agent'
      tag: 'Documentation'
      text: 'Getting Started with Datadog Agent'
---

{{< beta-callout url="#" btn_hidden="true" >}}
Autocomplete and search are in public beta.
{{< /beta-callout >}}

## Requirements

- [Dynamic Instrumentation][1] is enabled for your service.
- Tracing library [`dd-trace-dotnet`][1] 2.53.0 or higher is installed.

## Installation

Run your service with Dynamic Instrumentation enabled, and additionally enable autocomplete and search:

1. Set the `DD_SYMBOL_DATABASE_UPLOAD_ENABLED` environment variable to `true`.
1. Specify the `DD_SERVICE` and `DD_VERSION` [Unified Service Tags][2] as environment variables.
1. After starting your service with Dynamic Instrumentation with autocomplete and search enabled, you can use Dynamic Instrumentation's IDE-like features on the [APM > Dynamic Instrumentation page][3].

## Additional configuration

### Third party detection

> If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with these settings:

Author: @OmerRaviv, @EWachnowezki, what do you think? I think we should put the next version (2.54) as required. Otherwise we have to explain that the config settings only work from that version and so on.

> If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with these settings:

Reviewer 2: That sounds good to me!

---

> title: Enable autocomplete and search for .NET

Reviewer 3: ```suggestion
title: Enable Autocomplete and Search for .NET
```

kind: Documentation
aliases:
    - /tracing/dynamic_instrumentation/enabling/dotnet/

> is_beta: false

Reviewer 3: Should this be true like for Java?

private: false
code_lang: dotnet
type: multi-code-lang
code_lang_weight: 30

> further_reading:
>     - link: 'agent'
>       tag: 'Documentation'
>       text: 'Getting Started with Datadog Agent'

Reviewer 3: ```suggestion
```
You have to include the following at the end of the page for the further reading links to display:

```
## Further reading
 
{{< partial name="whats-next/whats-next.html" >}}
```

---

{{< beta-callout url="#" btn_hidden="true" >}}
Autocomplete and search are in public beta.
{{< /beta-callout >}}

## Requirements

- [Dynamic Instrumentation][1] is enabled for your service.
- Tracing library [`dd-trace-dotnet`][1] 2.54.0 or higher is installed.

## Installation

Run your service with Dynamic Instrumentation enabled, and additionally enable autocomplete and search:

1. Set the `DD_SYMBOL_DATABASE_UPLOAD_ENABLED` environment variable to `true`.
1. Specify the `DD_SERVICE` and `DD_VERSION` [Unified Service Tags][2] as environment variables.

> 1. After starting your service with Dynamic Instrumentation with autocomplete and search enabled, you can use Dynamic Instrumentation's IDE-like features on the [APM > Dynamic Instrumentation page][3].

Reviewer 3: ```suggestion
1. After starting your service with Dynamic Instrumentation and autocomplete and search enabled, you can use Dynamic Instrumentation's IDE-like features on the [**APM** > **Dynamic Instrumentation**][3] page.
```

## Additional configuration

> ### Third party detection

Reviewer 3: ```suggestion
### Third-party detection
```

> Autocomplete and search use a heuristic to filter out third-party code. If it does not provide the autocomplete suggestions you are expecting, it may be due to your code being recognized as third-party.

Reviewer 3: ```suggestion
If autocomplete suggestions do not appear for your package or module, it may be incorrectly recognized as third-party code. The autocomplete and search features use a heuristic to filter out third-party code, which can sometimes lead to accidental misclassification.
```

> If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with these settings:

Reviewer 3: ```suggestion
If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with the following settings:
```

> If your code is accidentally recognized as third-party code or third-party code is accidentally recognized as user code, you can configure the third-party detection with these settings:

Reviewer 3: ```suggestion
To ensure that your code is properly recognized and to enable accurate autocomplete and search functionality, you can configure the third-party detection settings using the following options:
```
